### PR TITLE
Use new references api

### PR DESCRIPTION
**Pull request by scharrier**
Since we have released our new references API, the legacy (and a little buggy) way of importing external references doesn't make sense anymore.

Previously, we had to parse all external references, transform them into hashes, import them in the original definition (inside some `x-imported/{n}` paths) which was then re-serialized and sent as a unique file to our servers.

With the new references API, all we have to do is to traverse the definition, identify all the external references, read them (as strings) and send them to our servers in the `references` body attribute.

This simplifies a lot our code, and make it far more stronger. Next steps are:
- enable by default the references import, as it's what our users want
- support recursive references (we still have some work to do server side for that)

### CODE REVIEW

File: lib/bump/cli/commands/base.rb [hunk 1]
```diff
@@ -18,9 +18,12 @@ def post(url:, body:, token: nil)
         def body(file, **options)
           deprecation_warning(options)
 
+          definition = Definition.new(file, import_external_references: options[:'import-external-references'])
+          definition.prepare!
+
           compact(
             {
-              definition: prepare_file(file, options),
+              definition: definition.content,
```
Comment (Polo2) [inline lib/bump/cli/commands/base.rb:26]: With this change, `prepare_file` become deadcode. Either if I miss something (review still in progress), method should be removed, and specs should be cleaned:
> `allow(command).to receive(:prepare_file).and_return("body")`

File: lib/bump/cli/commands/base.rb [hunk 2]
```diff
@@ -24,15 +24,18 @@ def body(file, **options)
...
               definition: definition.content,
+              references: prepare_references(definition.external_references),
               specification: options[:specification],
               validation: options[:validation],
               auto_create_documentation: options[:'auto-create']
             }.merge(documentation_or_hub(options))
           )
         end
 
-        def prepare_file(file, options)
```
Comment (Polo2) [inline lib/bump/cli/commands/base.rb:31]: I think this removal should belong to first commit (4da6e48), cf related comment

File: spec/bump/parser_spec.rb [hunk 3]
```diff
@@ -3,23 +3,20 @@
...
-      format, definition = Bump::CLI::Parser.new.load("property: value")
+      definition = Bump::CLI::Parser.new.load("property: value")
 
-      expect(format).to eq(:yaml)
       expect(definition).to eq("property" => "value")
     end
 
     it "loads JSON definition and serializes it correctly" do
-      format, definition = Bump::CLI::Parser.new.load('{"property": "value"}')
+      definition = Bump::CLI::Parser.new.load("{\"property\": \"value\"}")
```
Comment (Polo2) [inline spec/bump/parser_spec.rb]: Same here, it's strange that codestyle commit e53d626 didn't fixed it.

File: spec/bump/commands/validate_spec.rb [hunk 4]
```diff
@@ -3,17 +3,18 @@
 describe Bump::CLI::Commands::Validate do
   it "calls the server and exit successfully if success" do
     stub_bump_api_validate("validations/post_success.http")
+    allow(URI).to receive(:open).and_return(spy(read: "content"))
 
     expect {
       new_command.call(
         doc: "aaaa0000-bb11-cc22-dd33-eeeeee444444",
-        'doc-name': "Hello world",
+        "doc-name": "Hello world",
```
Comment (Polo2) [inline spec/bump/commands/validate_spec.rb:11]: I was surprised in codestyle commit e53d626, that this was not automatically handled by `standardrb`

File: lib/bump/cli/resource.rb [hunk 5]
```diff
@@ -3,8 +3,8 @@
 
 module Bump
   class CLI
-    class Parser
-      def load(content)
+    class Resource
```
Comment (Polo2) [inline lib/bump/cli/resource.rb:6]: Maybe the 3rd commit message should be updated, without mention to `File`

> * Rename Parser into File and #load into #parse
> * Move from instance method to class method
> * Rename File into Resource

**Review by Polo2 — APPROVED** [review]
Well done for this PR, a great improvement to support external references.

This new behavior is indeed better and easier to understand, I just noticed some deadcode that could be removed, and a commit message that should be updated.

Well done about the huge rebase on e53d626, it was a bit surprising that some codestyle you fix here were not automatically handled by this commit.

File: lib/bump/cli/commands/base.rb (hunk 1 above)
Comment (scharrier) [reply lib/bump/cli/commands/base.rb:26]: Nice catch, thanks! Fixed in a huge (but useful) rebase!

File: lib/bump/cli/commands/base.rb (hunk 2 above)
Comment (scharrier) [reply lib/bump/cli/commands/base.rb:31]: :+1:

File: spec/bump/commands/validate_spec.rb (hunk 4 above)
Comment (scharrier) [reply spec/bump/commands/validate_spec.rb:10]: Yes, and I just saw that there are still some issues. I will run it another time and manually fix what needs to be fixed.

File: lib/bump/cli/resource.rb (hunk 5 above)
Comment (scharrier) [reply lib/bump/cli/resource.rb:6]: Done, thanks :)

**Pull request merged**